mbf_costmap_nav/src/mbf_costmap_nav/costmap_navigation_server.cpp

```diff
@@ -825,4 +816,82 @@ bool CostmapNavigationServer::callServiceClearCostmaps(std_srvs::Empty::Request
   return true;
 }
 
+CostmapWrapper::Ptr CostmapNavigationServer::selectingRequestedCostmap(std::uint8_t costmap_type) const
+{
+  // selecting the requested costmap
+  CostmapWrapper::Ptr costmap;
+  switch (costmap_type)
+  {
+    case mbf_msgs::CheckPose::Request::LOCAL_COSTMAP:
+      return local_costmap_ptr_;
+      break;
+    case mbf_msgs::CheckPose::Request::GLOBAL_COSTMAP:
+      return global_costmap_ptr_;
+      break;
+    default:
+      ROS_ERROR_STREAM("No valid costmap provided; options are "
+                       << mbf_msgs::CheckPose::Request::LOCAL_COSTMAP << ": local costmap, "
+                       << mbf_msgs::CheckPose::Request::GLOBAL_COSTMAP << ": global costmap");
+      return costmap;
+  }
+}
+
+bool CostmapNavigationServer::callServiceFindValidPose(mbf_msgs::FindValidPose::Request& request,
+                                                       mbf_msgs::FindValidPose::Response& response)
+{
+  CostmapWrapper::Ptr costmap = selectingRequestedCostmap(request.costmap);
+  if (!costmap)
+  {
+    return false;
+  }
+  std::string costmap_name =
+      request.costmap == mbf_msgs::FindValidPose::Request::LOCAL_COSTMAP ? "local costmap" : "global costmap";
+
+  // get target pose or current robot pose as x, y, yaw coordinates
+  std::string costmap_frame = costmap->getGlobalFrameID();
+
+  geometry_msgs::PoseStamped pose;
+  if (request.current_pose)
+  {
+    if (!mbf_utility::getRobotPose(*tf_listener_ptr_, robot_frame_, costmap_frame, ros::Duration(0.5), pose))
+    {
+      ROS_ERROR_STREAM("Get robot pose on " << costmap_name << " frame '" << costmap_frame << "' failed");
+      return false;
+    }
+  }
+  else
+  {
+    if (!mbf_utility::transformPose(*tf_listener_ptr_, costmap_frame, ros::Duration(0.5), request.pose, pose))
+    {
+      ROS_ERROR_STREAM("Transform target pose to " << costmap_name << " frame '" << costmap_frame << "' failed");
+      return false;
+    }
+  }
+
+  geometry_msgs::Pose2D goal;
+  goal.x = request.pose.pose.position.x;
+  goal.y = request.pose.pose.position.y;
+  goal.theta = tf::getYaw(request.pose.pose.orientation);
+
+  ros::NodeHandle private_nh("~");
+  // using 5 degrees as increment
+  SearchConfig config{ ANGLE_INCREMENT,        request.angle_tolerance,
+                       request.dist_tolerance, static_cast<bool>(request.use_padded_fp),
+                       request.safety_dist,    goal };
+  SearchHelperViz viz(private_nh, costmap_frame);
+  SearchHelper search_helper(costmap.get(), config, std::nullopt, viz);
+
+  // search for a valid pose
+  const auto sol = search_helper.search();
+
+  response.pose.pose.position.x = sol.pose.x;
+  response.pose.pose.position.y = sol.pose.y;
+  response.pose.pose.orientation = tf::createQuaternionMsgFromYaw(sol.pose.theta);
```

Comment:
This returns consistently 0 yaw as orientation:
in
```
header: 
  seq: 0
  stamp: 0.000000000
  frame_id: map
pose: 
  position: 
    x: 2.9889
    y: 9.97502
    z: 0
  orientation: 
    x: 0
    y: 0
    z: 0.975985
    w: 0.217839
```
out
```
header: 
  seq: 0
  stamp: 1681505298.031356999
  frame_id: map
pose: 
  position: 
    x: 2.725
    y: 9.725
    z: 0
  orientation: 
    x: 0
    y: 0
    z: 0
    w: 1
```

and a video

https://user-images.githubusercontent.com/322610/232153951-a353d56d-5e27-4ab7-abf0-56f29c5b7f0c.mp4

Reply:
I have added a test for that:
https://github.com/magazino/move_base_flex/pull/317/files#diff-8fb727cf2dfe977e499637719122f6e082cda1178d701aa02d0c0d45a48d2204R393

The unittest works. Maybe I am missing something. Can you share how can I reproduce the issue?

Reply:
standalone navigation + clicked point nav goals, as in the video; nothing special 

Or just calling the service with any non-zero rotation returns a 0-yaw pose:
```
$ rosservice call /move_base_flex/find_valid_pose "pose:
  header:
    seq: 0
    stamp: {secs: 0, nsecs: 0}
    frame_id: 'map'
  pose:
    position: {x: 3.0, y: 6.0, z: 0.0}
    orientation: {x: 0.0, y: 0.0, z: 0.707, w: 0.707}
safety_dist: 0.0
dist_tolerance: 1.0
angle_tolerance: 0.0
costmap: 1
current_pose: false
use_padded_fp: false" 
```
gives on standalone nav
```
state: 1
cost: 253
pose: 
  header: 
    seq: 0
    stamp: 
      secs: 489
      nsecs: 450000000
    frame_id: "map"
  pose: 
    position: 
      x: 2.9875000000000043
      y: 6.212500000000004
      z: 0.0
    orientation: 
      x: 0.0
      y: 0.0
      z: 0.0
      w: 1.0
```

Reply:
Replacing initialization_list [here](https://github.com/magazino/move_base_flex/pull/317/files#diff-43fb1b9fdaf229fde2a9661293a0da41b5bfba67e7843bdb79dd19ce01fdd3cbR105) with vector solves the issue